sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -361,3 +361,17 @@ TEST(Context, Deadline)
     EXPECT_EQ(childCtx.GetDeadline(), Azure::DateTime::min());
   }
 }
+
+TEST(Context, ValueOr)
+{
+  const char* str = "Test String";
+  std::string s(str);
+
+  Context context;
+  Context::Key const key;
+
+  auto value = context.ValueOr<std::string>(key, str);
```

Comment:
This is testing the default value of `ValueOr`. Add a test for when the key is found as well, when calling `ValueOr`, to improve branch and line coverage.

Reply:
Also please add a test for returning a const& if that's what the method returns